sdk/core/azure-core/azure/core/async_paging_method.py

```diff
@@ -0,0 +1,234 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from typing import Any, Callable, Iterable, Tuple
+from .paging import ResponseType, ReturnType
+from ._pipeline_client import PipelineClient
+from .pipeline.transport import HttpRequest, HttpResponse
+
+from .exceptions import (
+    HttpResponseError, ClientAuthenticationError, ResourceExistsError, ResourceNotFoundError, map_error
+)
+
+class AsyncPagingMethodABC:
```

Comment:
Why didn't we make this into a "real" abstract class? See https://six.readthedocs.io/#six.add_metaclass and https://docs.python.org/3.2/library/abc.html#abc.ABCMeta for more info on how to make it py2 compatible....

Reply:
..which also begs the question, is this intended to be py3 only code?